File: algorithms/max_subset_sum_no_adjacent.py
def maxSubsetSumNoAdjacent(array):
    curr_max = 0
    prev_max = 0
    for i in range(len(array)):
    	temp = curr_max
    	curr_max = max(prev_max + array[i], curr_max)
    	prev_max = temp
	
    return curr_max

File: algorithms/javascript/max_subset_sum_no_adjacent.js
function maxSubsetSumNoAdjacent(array) {
    let current_max = 0;
    let last_max = 0;
    let prev_last_max = 0;

    for (let i = 0; i < array.length; i++) {
        prev_last_max = last_max;
        last_max = current_max;
        current_max = Math.max(prev_last_max + array[i], current_max)
    }

    return current_max;
}